src/modules/Dashboard/features/Applet/Schedule/ConfirmScheduledAccessPopup/ConfirmScheduledAccessPopup.test.tsx

```diff
@@ -0,0 +1,51 @@
+import { fireEvent, screen } from '@testing-library/react';
+
+import { renderWithProviders } from 'shared/utils';
+
+import { ConfirmScheduledAccessPopup } from './ConfirmScheduledAccessPopup';
+
+const onCloseMock = jest.fn();
+const onSubmitMock = jest.fn();
+
+describe('ConfirmScheduledAccessPopup', () => {
+  test('should render and submit', () => {
+    renderWithProviders(
+      <ConfirmScheduledAccessPopup
+        open={true}
+        onClose={onCloseMock}
+        onSubmit={onSubmitMock}
+        isLoading={false}
+        activityName="New Activity"
+      />,
+    );
+
+    const popup = screen.getByTestId('dashboard-calendar-confirm-scheduled-access-popup');
+    expect(popup).toBeInTheDocument();
+    expect(popup).toHaveTextContent(
+      'Activity New Activity will no longer be always available, and the Activity will be a scheduled event. Are you sure you want to continue?',
+    );
+
+    fireEvent.click(screen.getByText('Confirm'));
+
+    expect(onSubmitMock).toBeCalled();
+  });
+
+  test('should show spinner and disable submit when isLoading true', () => {
+    renderWithProviders(
+      <ConfirmScheduledAccessPopup
+        open={true}
+        onClose={onCloseMock}
+        onSubmit={onSubmitMock}
+        isLoading={true}
+        activityName="New Activity"
+      />,
+    );
+
+    expect(screen.getByTestId('spinner')).toBeInTheDocument();
```

Comment:
it doesn't make sense to expect elements got from getBy* methods being in the document, because the result of these methods is the element found in the DOM, so it does make sense to check toBeVisible for example